### PR TITLE
DOP-4479: Update github actions

### PR DESCRIPTION
**Pull request by i80and**
### Ticket

DOP-4479

### Notes

Test workflow succeeded in https://github.com/mongodb/snooty-parser/actions/runs/8573383377. Confirmed that a release was made with both assets attached.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README.md and/or HACKING.md, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README.md and/or HACKING.md

### CODE REVIEW

**Review by rayangler — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -40,17 +40,19 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Download
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
-          name: packages
+          path: packages
+          pattern: package-*
```
Comment (rayangler) [inline .github/workflows/release.yml:46]: Oh, actually, I have a question: Will the [path the Autobuilder uses](https://github.com/mongodb/docs-worker-pool/blob/24ec3e5c0a4b4f6c6a421862680d095063d4f816/Dockerfile#L51) to download the packages need to be updated?

Comment (i80and) [reply .github/workflows/release.yml:46]: Should just work! The zip file names aren't changing, and they look to be what I expect in the test pseudo-release I did*.

* which is slightly different from a real release, but only in a way that *shouldn't* matter.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,9 +29,9 @@ jobs:
       id: build_package
       run: make package
     - name: Upload package
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
```
Comment (seungpark) [inline .github/workflows/release.yml:32]: (question) is there a reason we are building the parser in the test workflow (vs the release workflow)?

**Review by seungpark — APPROVED** [review]
this LGTM. thanks for providing the build log! wondering how you triggered a release job without pushing a tag?

File: .github/workflows/release.yml (hunk 2 above)
Comment (i80and) [reply .github/workflows/release.yml:32]: The test workflow runs per OS in the matrix, while the release workflow only runs once and merges the resulting artifacts together. Our test processes do a build anyway to make sure the binaries continue to work

**Comment by i80and** [comment]
> this LGTM. thanks for providing the build log! wondering how you triggered a release job without pushing a tag?

I added the branch to the condition for the workflow, and commented out the conditional for it needing to start with `v` 😹

**Pull request merged**